### PR TITLE
Fix/broken nav categories #860

### LINKED ISSUE #860: [BUG] Menu is Broken on Demo Shop
### Describe the bug
If you have many menu items you can not select the correct sub menu

- go to homepage
- hover a menu element

### Additional context
menu

### PR DESCRIPTION
**Pull request by Malwurf**
## Changes
- Added categories aggregation to handle more than 3-4 categories in the top navbar
- Fixed losing hover on mouse out event of a category name
- Added some global styles to nav categories

Closes #860 and #853 

### Checklist

- [x] I followed [contributing](https://github.com/DivanteLtd/shopware-pwa/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

**Comment by akrajna** [comment]
well done :-)

File: packages/default-theme/components/SwTopNavigation.vue [hunk 1]
```diff
@@ -33,6 +46,7 @@ import { ref, onMounted, watch } from "@vue/composition-api"
 import { getCategoryUrl } from "@shopware-pwa/helpers"
 import SwPluginSlot from "sw-plugins/SwPluginSlot"
 import { useLocales } from "@shopware-pwa/default-theme/logic"
+import { setTimeout } from "timers"
```
Comment (patzick) [inline packages/default-theme/components/SwTopNavigation.vue:49]: :)
```suggestion
```

File: packages/default-theme/components/SwTopNavigation.vue [hunk 2]
```diff
@@ -13,14 +13,27 @@
...
     </SfHeaderNavigationItem>
+
+    <SfHeaderNavigationItem
+      class="sf-header__link"
+      v-if="unvisibleCategories"
+      @mouseover="changeCurrentCategory('categories')"
+      @mouseleave="changeCurrentCategory(null)"
+      @keyup.tab="changeCurrentCategory('categories')"
+    >
+      <span class="sf-header__link">More</span>
```
Comment (patzick) [inline packages/default-theme/components/SwTopNavigation.vue:31]: Would be great to use a new small component here like `SwTopNavigationShowMore` or something like this, then if someone wants to add some icon or change appearance, then only a small component need to be changed :)

Also `More` text should be translatable, please add English and German version

Comment (Malwurf) [reply packages/default-theme/components/SwTopNavigation.vue]: Thanks, I moved that span to separate component and added translations in EN and DE.

**Pull request merged**